tests/uri_router_test.py

```diff
@@ -9,6 +9,39 @@
 pytest.importorskip('srwl_bl')
 
 
+def test_error_for_bots():
+    from pykern import pkcompat
+    from pykern.pkcollections import PKDict
+    from pykern.pkunit import pkeq, pkexcept, pkok, pkre
+    from sirepo import http_request
+    from sirepo import srunit
+
+    fc = srunit.flask_client()
+    fc.sr_login_as_guest()
+
+    uri = '/get-application-data'
+    d = PKDict(simulationType='srw', method='NO SUCH METHOD')
+
+    # "Real" browsers get redirected to an error page with a 200 status
+    r = fc.post(
+        uri,
+        environ_base=PKDict(HTTP_USER_AGENT=http_request.USER_AGENT_MOZILLA),
+        json=d
+    )
+    pkeq(200, r.status_code)
+    pkok(
+        '/srw#/error' in pkcompat.from_bytes(r.data),
+        'Unknown data method should redirect to error page'
+    )
+
+    r = fc.post(
+        uri,
+        environ_base=PKDict(HTTP_USER_AGENT=http_request.USER_AGENT_PYTHON),
```

Comment:
Here we want a specific version, e.g. python-requests/2.0 and maybe another version python-requests/1.3.

Reply:
1164018